pkg/cmd/server/startconfig/start_allinone.go

```diff
@@ -0,0 +1,207 @@
+integrated instance, or --kubernetes=<addr> and --kubeconfig=<path> to connect to an existing
+Kubernetes cluster.
+`
+
+// NewCommandStartMaster provides a CLI handler for 'start' command
+func NewCommandStartAllInOne() (*cobra.Command, *AllInOneOptions) {
+	options := &AllInOneOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Launch OpenShift All-In-One",
+		Long:  longAllInOneCommandDesc,
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(); err != nil {
+				glog.Fatal(err)
+			}
+			if err := options.Validate(); err != nil {
+				glog.Fatal(err)
+			}
+
+			if err := options.StartAllInOne(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.BoolVar(&options.WriteConfigOnly, "config-only", false, "Indicates that the command should build the config that would be used to start OpenShift and do nothing else.")
+	flags.StringVar(&options.MasterConfigFile, "master-config", "", "Indicates that the command use the config file from a certain location.  If this is specified, all other options are ignored except --config-only.")
+	flags.StringVar(&options.NodeConfigFile, "node-config", "", "Indicates that the command use the config file from a certain location.  If this is specified, all other options are ignored except --config-only.")
+
+	masterArgs, nodeArgs, bindAddrArg, imageFormatArgs, kubeConnectionArgs, certArgs := GetAllInOneArgs()
+	options.MasterArgs, options.NodeArgs = masterArgs, nodeArgs
+
+	BindMasterArgs(masterArgs, flags, "")
+	BindNodeArgs(nodeArgs, flags, "")
+	BindBindAddrArg(bindAddrArg, flags, "")
+	BindImageFormatArgs(imageFormatArgs, flags, "")
+	BindKubeConnectionArgs(kubeConnectionArgs, flags, "")
+	BindCertArgs(certArgs, flags, "")
+
+	startMaster, _ := NewCommandStartMaster()
+	startNode, _ := NewCommandStartNode()
+	cmd.AddCommand(startMaster)
+	cmd.AddCommand(startNode)
+
+	return cmd, options
+}
+
+// GetAllInOneArgs makes sure that the node and master args that should be shared, are shared
+func GetAllInOneArgs() (*MasterArgs, *NodeArgs, *BindAddrArg, *ImageFormatArgs, *KubeConnectionArgs, *CertArgs) {
+	masterArgs := NewDefaultMasterArgs()
+	nodeArgs := NewDefaultNodeArgs()
+
+	bindAddrArg := NewDefaultBindAddrArg()
+	masterArgs.BindAddrArg = bindAddrArg
+	nodeArgs.BindAddrArg = bindAddrArg
+
+	imageFormatArgs := NewDefaultImageFormatArgs()
+	masterArgs.ImageFormatArgs = imageFormatArgs
+	nodeArgs.ImageFormatArgs = imageFormatArgs
+
+	kubeConnectionArgs := NewDefaultKubeConnectionArgs()
+	masterArgs.KubeConnectionArgs = kubeConnectionArgs
+	nodeArgs.KubeConnectionArgs = kubeConnectionArgs
+
+	certArgs := NewDefaultCertArgs()
+	masterArgs.CertArgs = certArgs
+	nodeArgs.CertArgs = certArgs
+	kubeConnectionArgs.CertArgs = certArgs
+
+	return masterArgs, nodeArgs, bindAddrArg, imageFormatArgs, kubeConnectionArgs, certArgs
+}
+
+func (o AllInOneOptions) Validate() error {
+	return nil
```

Comment:
`openshift start foo` doesn't complain... should prevent additional args we ignore